pleskdistup/actions/distupgrade.py

```diff
@@ -145,6 +147,118 @@ def estimate_revert_time(self) -> int:
         return 20
 
 
+class SetupAptReposRegexp(action.ActiveAction):
+    from_regexp: str
+    to_regexp: str
+    sources_list_path: str
+    sources_list_d_path: str
+    backup_suffix: str
+    _name: str
+
+    def __init__(
+        self,
+        from_regexp: str,
+        to_regexp: str,
+        sources_list_path: str = "/etc/apt/sources.list",
+        sources_list_d_path: str = "/etc/apt/sources.list.d/",
+        name: str = "set up APT repositories to change from {self.from_regexp!r} to {self.to_regexp!r}",
+    ):
+        self.from_regexp = from_regexp
+        self.to_regexp = to_regexp
+        self.sources_list_path = sources_list_path
+        self.sources_list_d_path = sources_list_d_path
+
+        self._name = name
+        self.backup_suffix = "_" + str(uuid.uuid4())
+
+    @property
+    def name(self):
+        return self._name.format(self=self)
+
+    def _apply_replace_on_file(self, fpath: str, ptrn: re.Pattern, to_regexp: str) -> None:
+        changed = False
+        new_lines = []
+        with open(fpath) as f:
+            for line in f:
+                new_lines.append(ptrn.sub(to_regexp, line))
+                if new_lines[-1] != line:
+                    changed = True
+        if not changed:
+            return
+        with open(fpath + self.backup_suffix, 'wb') as b:
```

Comment:
I suggest using [`files.backup_file`](https://github.com/plesk/dist-upgrader/blob/main/pleskdistup/common/src/files.py#L62-L64) to back up files.

Reply:
I utilized it by changing a bit to match my logic. I didn't want to use some common `.bak` extensions so I produced my UUID.

Reply:
And I lost the ability to move my UUID suffixed temporary backups to some well known extension like ".distupg_save" at the end, just in case.